third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/ScanOpToLLVM.cpp

```diff
@@ -462,6 +482,11 @@ ScanOpConversion::emitFastScan(triton::ScanOp op, triton::ScanOpAdaptor adaptor,
   auto srcValues =
       unpackInputs(loc, op, adaptor, rewriter, *getTypeConverter());
 
+  if (op.getReverse()) {
+      warpIdAxis = sub(i32_val(axisNumWarps-1), warpIdAxis);
+      srcValues = flipSrcValues(loc, op, rewriter, srcValues, iWarpSize);
```

Comment:
Would it make sense to make this a lowering for `tl.flip` and implement the reverse scan in python?

Reply:
That makes sense. Wasn't sure how to do that efficiently though. Is there code you would recommend looking at?

Reply:
Actually, now I understand the reversing the `warpIdAxis` I think this does make more sense to be coupled to the scan op.